t/cve-2017-3302.t

```diff
@@ -15,10 +9,11 @@ use vars qw($test_dsn $test_user $test_password $test_db);
 use lib 't', '.';
 require "lib.pl";
 eval {
-  $dbh = DBI->connect($test_dsn, $test_user, $test_password, {RaiseError => 1});
+  $dbh = DBI->connect($test_dsn, $test_user, $test_password, {RaiseError => 1, mysql_server_prepare => 1});
+  die 'MySQL version less than 4.1 does not support prepared statements' unless MinimumVersion($dbh, '4.1');
```

Comment:
Is it necessary though? I'm not super familiar with C, but it seems to me that version checks are already in place at compile time for the client versions. If the client doesn't support prepared statements, then the attributes aren't even applied.

https://github.com/perl5-dbi/DBD-mysql/blob/master/dbdimp.c#L2039-L2067